libraries/botframework-connector/tests/skillValidation.test.js

```diff
@@ -145,49 +145,60 @@ describe('SkillValidation', function () {
+        const makeCredentialsAndIdentity = ({
+            audience,
+            audienceClaim, // overrides audience in claim value
+            appId = uuid(),
+            authType = true,
+            version = '1.0',
+        } = {}) => {
+            if (audience === undefined) {
+                audience = appId;
+            }
 
-        const appIdClaim = { type: AuthenticationConstants.AppIdClaim, value: appId };
-        const audClaim = { type: AuthenticationConstants.AudienceClaim, value: audience };
-        const verClaim = { type: AuthenticationConstants.VersionClaim, value: '1.0' };
+            const credentials = new SimpleCredentialProvider(audience, '');
 
-        const claims = [appIdClaim, audClaim, verClaim];
-        const identity = new ClaimsIdentity(claims);
+            const appIdClaim = { type: AuthenticationConstants.AppIdClaim, value: appId };
+            const audClaim = {
+                type: AuthenticationConstants.AudienceClaim,
+                value: audienceClaim !== undefined ? audienceClaim : audience,
+            };
+            const verClaim = { type: AuthenticationConstants.VersionClaim, value: version };
 
-        afterEach(() => {
-            identity.isAuthenticated = true;
+            const claims = [appIdClaim, audClaim, verClaim];
+            const identity = new ClaimsIdentity(claims, authType);
 
-            verClaim.type = AuthenticationConstants.VersionClaim;
-            verClaim.value = '1.0';
-            audClaim.type = AuthenticationConstants.AudienceClaim;
-            audClaim.value = audience;
-            appIdClaim.type = AuthenticationConstants.AppIdClaim;
-            appIdClaim.value = appId;
-        });
+            return { credentials, identity };
+        };
 
         it('should fail with a falsey identity', async () => {
+            const { credentials } = makeCredentialsAndIdentity();
+
             try {
                 await SkillValidation.validateIdentity(undefined, credentials);
+                assert.fail('should have thrown');
             } catch (e) {
                 assert.strictEqual(e.message, 'SkillValidation.validateIdentity(): Invalid identity');
             }
         });
 
         it('should fail if ClaimsIdentity.isAuthenticated is false', async () => {
+            const { credentials, identity } = makeCredentialsAndIdentity({ authType: false });
+
             try {
-                identity.isAuthenticated = false;
                 await SkillValidation.validateIdentity(identity, credentials);
+                assert.fail('should have thrown');
             } catch (e) {
                 assert.strictEqual(e.message, 'SkillValidation.validateIdentity(): Token not authenticated');
             }
         });
 
-        it('should fail no version claim', async () => {
+        it('should fail with no version claim', async () => {
+            const { credentials, identity } = makeCredentialsAndIdentity({ version: null });
```

Comment:
Looking at this makes me wonder why we didn't populate a "full" fake ClaimsIdentity for no-auth Skills.

E.g. the ClaimsIdentity.claims looks like this:

```ts
[
  { type: AuthenticationConstants.VersionClaim, value: '2.0' },
  { type: AuthenticationConstants.AuthorizedParty, value: AuthenticationConstants.AnonymousBotCaller},
  { type: AuthenticationConstants.AudienceClaim, value: AuthenticationConstants.AnonymousSkillId }
]
```

This isn't a comment on this specific test, but rather the no-auth skills implementation.

Reply:
We certainly could, I think the ClaimsIdentity used for no-auth skills is purely a placeholder. If nothing else, it might lure us into using that set of claims for other things.

Reply:
True, I think utilizing the ClaimsIdentity for this has led to some confusion and perpetuate some problems. But the current no-auth Skills ClaimsIdentity is incomplete and I believe part of the reason we had a few conversations about when to not set a token, etc.